### PR TITLE
[net8.0] Merge latest main to net8

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

Brings latest changes on main to the net8.0 branch for next release.

### CODE REVIEW

File: src/Core/src/PublicAPI/net-android/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -2206,7 +2206,6 @@ override Microsoft.Maui.MauiAppCompatActivity.OnActivityResult(int requestCode,
 override Microsoft.Maui.MauiAppCompatActivity.OnBackPressed() -> void
 override Microsoft.Maui.MauiAppCompatActivity.OnConfigurationChanged(Android.Content.Res.Configuration! newConfig) -> void
 override Microsoft.Maui.MauiAppCompatActivity.OnCreate(Android.OS.Bundle? savedInstanceState) -> void
-override Microsoft.Maui.MauiAppCompatActivity.OnDestroy() -> void
```
Comment (rmarinho) [inline src/Core/src/PublicAPI/net-android/PublicAPI.Shipped.txt:2209]: @mattleibow shouldn't this be marked as remove on Unshipped ?

File: src/Core/src/PublicAPI/net-android/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -37,6 +44,13 @@ override Microsoft.Maui.SizeRequest.GetHashCode() -> int
...
+static Microsoft.Maui.CommandMapperExtensions.AppendToMapping<TVirtualView, TViewHandler>(this Microsoft.Maui.ICommandMapper<Microsoft.Maui.IElement!, Microsoft.Maui.IElementHandler!>! commandMapper, string! key, System.Action<TViewHandler, TVirtualView, object?>! method) -> void
+static Microsoft.Maui.CommandMapperExtensions.ModifyMapping<TVirtualView, TViewHandler>(this Microsoft.Maui.ICommandMapper<TVirtualView, TViewHandler>! commandMapper, string! key, System.Action<TViewHandler, TVirtualView, object?, System.Action<Microsoft.Maui.IElementHandler!, Microsoft.Maui.IElement!, object?>?>! method) -> void
+static Microsoft.Maui.CommandMapperExtensions.ModifyMapping<TVirtualView, TViewHandler>(this Microsoft.Maui.ICommandMapper<Microsoft.Maui.IElement!, Microsoft.Maui.IElementHandler!>! commandMapper, string! key, System.Action<TViewHandler, TVirtualView, object?, System.Action<Microsoft.Maui.IElementHandler!, Microsoft.Maui.IElement!, object?>?>! method) -> void
```
Comment (rmarinho) [inline src/Core/src/PublicAPI/net-android/PublicAPI.Unshipped.txt:50]: @mattleibow but I m adding the api here too not sure why is complaining

**Review by Eilon — APPROVED** [review]

**Review by jsuarezruiz — APPROVED** [review]

**Comment by rmarinho** [comment]
@pjcollins  any idea why testes would be failing ?

File: src/Core/src/PublicAPI/net-android/PublicAPI.Shipped.txt (hunk 1 above)
Comment (mattleibow) [reply src/Core/src/PublicAPI/net-android/PublicAPI.Shipped.txt:2209]: It probably should have, but it is an override so we can let it slide...

File: src/Templates/src/templates/maui-mobile/MauiApp.1.csproj [hunk 3]
```diff
@@ -56,6 +56,8 @@
 	</ItemGroup>
 
 	<ItemGroup>
+		<PackageReference Include="Microsoft.Maui.Controls" Version="$(MauiVersion)" />
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply src/Templates/src/templates/maui-mobile/MauiApp.1.csproj:59]: This is new. When updating package refs in the IDE or CLI, the implicit packages causes issues. So we are now explicit. We also have a new warning for existing net8 projects. In net9 we may only do explicit packages as implicit packages is not 100% recommended by anyone.

**Review by jfversluis — APPROVED** [review]

**Pull request merged**